terraform/lb.tf

```diff
@@ -11,47 +11,108 @@
-#  provider = google-beta
-#  name     = "verification-server"
-#  project  = var.project
-#
-#  ip_protocol           = "TCP"
-#  ip_address            = google_compute_global_address.verification-server.address
-#  load_balancing_scheme = "EXTERNAL"
-#  port_range            = "80"
-#  target                = google_compute_target_http_proxy.default.id
-#  network_tier          = "PREMIUM"
-#}
+# Redirects all requests to https
+resource "google_compute_url_map" "urlmap-http" {
+  name     = "https-redirect"
+  provider = google-beta
+  project  = var.project
+
+  default_url_redirect {
+    strip_query    = false
+    https_redirect = true
+  }
+}
+
+resource "google_compute_url_map" "urlmap-https" {
+  name            = "verification-server"
+  provider        = google-beta
+  project         = var.project
+  default_service = google_compute_backend_service.apiserver.id
+
+  host_rule {
+    hosts        = [var.server-host]
+    path_matcher = "server"
+  }
+
+  path_matcher {
+    name            = "server"
+    default_service = google_compute_backend_service.server.id
+  }
+
+  host_rule {
+    hosts        = [var.apiserver-host]
+    path_matcher = "apiserver"
+  }
+
+  path_matcher {
+    name            = "apiserver"
+    default_service = google_compute_backend_service.apiserver.id
+  }
+
+  host_rule {
+    hosts        = [var.adminapi-host]
+    path_matcher = "adminapi"
+  }
+
+  path_matcher {
+    name            = "adminapi"
+    default_service = google_compute_backend_service.adminapi.id
+  }
+}
+
+resource "google_compute_target_http_proxy" "http" {
+  provider = google-beta
+  name     = "verification-server"
+  project  = var.project
+
+  url_map = google_compute_url_map.urlmap-http.id
+}
+
+resource "google_compute_target_https_proxy" "https" {
+  name    = "verification-server"
+  project = var.project
+
+  url_map          = google_compute_url_map.urlmap-https.id
+  ssl_certificates = [google_compute_managed_ssl_certificate.default.id]
+}
+
+resource "google_compute_forwarding_rule" "http" {
+  provider = google-beta
+  name     = "verification-server-http"
+  project  = var.project
+
+  ip_protocol           = "TCP"
+  ip_address            = google_compute_global_address.verification-server.address
+  load_balancing_scheme = "EXTERNAL"
+  port_range            = "80"
+  target                = google_compute_target_http_proxy.http.id
+  network_tier          = "PREMIUM"
+}
+
+resource "google_compute_forwarding_rule" "https" {
+  provider = google-beta
+  name     = "verification-server-https"
+  project  = var.project
+
+  ip_protocol           = "TCP"
+  ip_address            = google_compute_global_address.verification-server.address
+  load_balancing_scheme = "EXTERNAL"
+  port_range            = "443"
+  target                = google_compute_target_https_proxy.https.id
+  network_tier          = "PREMIUM"
+}
+
+resource "google_compute_managed_ssl_certificate" "default" {
```

Comment:
Hmm so how does this work? Cloud Run already gives us a TLS cert.

/cc @mikehelmick

Reply:
This is the SSL cert on the GCLB. Cloud Run will still have it's urls available.

From https://cloud.google.com/load-balancing/docs/negs/setting-up-serverless-negs, this is the path of the things:

![flow](https://cloud.google.com/load-balancing/images/lb-serverless-run.svg)

Reply:
For example, you can look at server.test.encv.org and look at it's ssl cert, it's generated through this method.

Reply:
I guess I'm worried about having two certs for the same URL.

Reply:
Technically that's not a problem, you can have as many certs for a domain as you want. 

With this change tho, we don't need any of the custom cloud run domain stuff tho, so that's one less thing to worry about.

Reply:
Just as proof, here's all the certs already issued for encv.org: https://transparencyreport.google.com/https/certificates?hl=en&cert_search_auth=&cert_search_cert=&cert_search=include_subdomains:true;domain:encv.org&lu=cert_search

Reply:
I'm more worried if an app is doing CA pinning...

Reply:
In what scenario? All of the certs we've issued are through Google's Let's Encrypt thing (GTS) which all expire every 3 months. This and cloud run use the same cert issuing service. We're also introducing new domains if they are doing some sort of weird pinning there. They could pick a CA further up the chain (their choices for all of our certs are GTS or GlobalSign), but those should be even more stable. I guess I just don't see this as an issue if they are doing cert pinning correctly.

Reply:
Okay fair. I'm probably just being TLScared.